File: tests/steps/product_detail.steps.ts
import { createBdd } from 'playwright-bdd';
import { test } from '../support/fixtures';

const { Given } = createBdd(test);

Given('I go to the {string} product page', async ({ product }, name: string) => {
  await product.viewProductPage(name);
});

Given('I add product to cart', async ({ product }, name: string) => {
  await product.addItemToBasket();
});

File: tests/steps/product_detail.steps.cs
using System.Threading.Tasks;
using TechTalk.SpecFlow;
using MyTests.Pages;

namespace MyTests.Steps;

[Binding]
public sealed class ProductSteps
{
    private readonly ProductDetailPage _product;

    public ProductSteps(ProductDetailPage product)
    {
        _product = product;
    }

    [Given(@"I go to the ""(.*)"" product page")]
    public async Task GivenIGoToTheProductPage(string name)
    {
        await _product.ViewProductPageAsync(name);
    }

    [Given(@"I add product to cart")]
    public async Task GivenIAddProductToCart()
    {
        await _product.AddItemToBasketAsync();
    }
}
